mitiq/rem/measurement_result.py

```diff
@@ -0,0 +1,54 @@
+# Copyright (C) 2021 Unitary Fund
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Defines MeasurementResult, a result obtained by measuring qubits on a
+quantum computer."""
+from dataclasses import dataclass
+from typing import Iterable, List
+
+import numpy as np
+
+
+Bitstring = List[int]
+
+
+@dataclass
+class MeasurementResult:
+    result: List[Bitstring]
+
+    def __post_init__(self) -> None:
+        self._bitstrings = np.array(self.result)
```

Comment:
What happens if `MeasurementResult` is initialized with something that is not a `List[List[int]]` ?   We may raise some error. If the input as an `np.array`, we could also try to cast it in the expected form (`List[List[int]]`).

Reply:
Good suggestion. Added support for this + a test.